In [2]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
%matplotlib inline

import seaborn as sns

from sklearn.tree import DecisionTreeClassifier as DT
from sklearn.model_selection import cross_validate
from sklearn.model_selection import GridSearchCV

In [3]:
rank = pd.read_csv("rankdata.csv")

In [10]:
rank.describe()

順位        暫定評価
count  666.000000  666.000000
mean   333.500000    0.827737
std    192.401923    0.126248
min      1.000000    0.000000
25%    167.250000    0.844758
50%    333.500000    0.862048
75%    499.750000    0.871200
max    666.000000    0.955347

暫定評価0.87を目標にする。→0.87達成、変えたことは決定木の作り方をDecisionTreeClassifierからLightGBM（より高性能なアルゴリズム）に変更したことくらい

他に工夫したこと：yについて0or1に変換したので予測値は0-1の間の小数で出た。それを<=50K<に戻す際にどこをボーダーにするべきかを自分で可視化した